server/control.go

```diff
@@ -227,6 +227,15 @@ func (c Controller) runOnce(ctx context.Context, leaderKey string, tick <-chan t
 
 	inf, err := cke.NewInfrastructure(ctx, cluster, storage)
 	if err != nil {
+		// When the vault token is revoked, the following error will be returned. In this case, CKE can not continue any operations.
+		// Error: "Error making API request.\n\nURL: GET <<URL>>\nCode: 403. Errors:\n\n* permission denied"
+		if strings.Contains(err.Error(), "Code: 403.") {
```

Comment:
I think checking "Code: 403.” in the error string has a risk of being outdated.
Why don't you check only "403"?

Reply:
That's good. I will change it.

It's a trade-off for false positives.
But even if CKE terminates due to a false positive, it won't cause any problems.